lib/lib.go

```diff
@@ -32,16 +45,417 @@ func init() {
 
 // Receivers returns a slice of CoreRequests that defines the full local
 // API of lib methods
-func Receivers(node *p2p.QriNode) []Requests {
+func Receivers(inst Instance) []Requests {
+	node := inst.Node()
+	r := inst.Repo()
+
 	return []Requests{
 		NewDatasetRequests(node, nil),
 		NewRegistryRequests(node, nil),
 		NewLogRequests(node, nil),
 		NewExportRequests(node, nil),
 		NewPeerRequests(node, nil),
-		NewProfileRequests(node, nil),
+		NewProfileMethods(inst),
+		NewConfigMethods(inst),
 		NewSearchRequests(node, nil),
-		NewRenderRequests(node.Repo, nil),
-		NewSelectionRequests(node.Repo, nil),
+		NewRenderRequests(r, nil),
+		NewSelectionRequests(r, nil),
+	}
+}
+
+// Instance is the interface that bundles the foundational values of a
+// qri instance. Instance provides the basis for creating Method constructors,
+// which actually do qri things
+// think of instance as the "core" of the qri ecosystem
+type Instance interface {
+	// Context returns the base context this instance is using. Any resources
+	// built from this instance should inherit from this context and obey
+	// calls from the ctx.Done(), releasing any & all resources
+	Context() context.Context
+	// Teardown closes the instance by closing the base context
+	Teardown()
+	// Config returns the current configuration for this
+	Config() *config.Config
+	Node() *p2p.QriNode
+	Repo() repo.Repo
+	RPC() *rpc.Client
+}
+
+// WritableInstance is an instance that can be modified
+// instances that allow config changes should implement this interface
+type WritableInstance interface {
```

Comment:
Discussed offline: I don't think it works as well having this property expressed as an interface. Would prefer to have explicit logic in the config code to control whether the config may be changed. Thinking about it even more, using the name `WritableInstance` also implies that nothing *about* the instance is writable, like it would make me think that the repo and node were also could not be written to if they didn't implement this interface.